### ISSUE TITLE
[Bug] Exception in console

### ISSUE THREAD

**Issue by ScrapMetalGolem** [Baktus79/BanFromClaim#46]
**Describe the bug**

Server recently updated to 1.21. At inconsistent intervals (minutes or hours apart), error message appears in console. Cause is unknown, and I really don't have a good lead on it yet.

---

**ERROR MESSAGE**
```11:59:34 [WARN] [BanFromClaim] Task #124 for BanFromClaim v1.7.2 generated an exception
java.lang.NullPointerException: Cannot invoke "com.griefdefender.api.claim.Claim.isWilderness()" because "claim" is null
        at Banfromclaim.jar/no.vestlandetmc.BanFromClaim.hooks.GriefDefenderHook.regionExist(GriefDefenderHook.java:104) ~[Banfromclaim.jar:?]
        at Banfromclaim.jar/no.vestlandetmc.BanFromClaim.config.ClaimData.cleanDatafile(ClaimData.java:173) ~[Banfromclaim.jar:?]
        at Banfromclaim.jar/no.vestlandetmc.BanFromClaim.BfcPlugin$1.run(BfcPlugin.java:76) ~[Banfromclaim.jar:?]
        at org.bukkit.craftbukkit.scheduler.CraftTask.run(CraftTask.java:86) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at org.bukkit.craftbukkit.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:475) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1762) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:509) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1634) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1334) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:333) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]```

---

**Additional server information**

**BanFromClaim version:1.7.2
**Server version (use /ver):Current Purpur Version: 1.21-2283-75e81d8 (MC: 1.21)
**Plugin list:**
Paper Plugins: (1):
- SignedVelocity
Bukkit Plugins: (54):
- 2FAGuard, AdvancedMobArena, AntiPopup, AuraSkills, BanFromClaim, BanManager, BattlePass, BigDoors, BlueMap, ChatControlRed
Citizens, CMI, CMIEInjector, CMILib, CoreProtect, DiscordSRV, DisplayShops, DragonSlayer, ExcellentChallenges, ExcellentCrates
ExtraContexts, FastAsyncWorldEdit, FastChunkPregenerator, FunGames, GDHooks, GriefDefender, GrimAC, HeadDatabase, IllegalStack, ItemFilter
Jobs, LibsDisguises, LuckPerms, MineResetLite, Multiverse-Core, nightcore, OreAnnouncer, packetevents, PinataParty, PlaceholderAPI
ProtocolLib, PurpurExtras, SafariNet, SternalBoard, Timedime, TradeMe, UltimateAutoRestart, UltraCosmetics, Vault, VoidGen
Votifier, WorldEditSUI, WorldGuard, WorldGuardExtraFlags

**Comment by Bicheka** [comment]
Could you explain a little more how to reproduce the error?

**Comment by ScrapMetalGolem** [comment]
Currently the server is idle with 0 players. I recently updated it to Purpur 1.21, deleted the overworld, and pregenerated the whole world. Then I removed all the old files for Bluemap and recreated new maps for all 3 worlds.

There were a few griefdefender claims in the old world (it's not public yet), and I haven't done anything to/with those since recreating the world. I doubt anyone is banned from any claims but I can check that later.

Interesting that the two error messages since my last restart occurred at exactly 11:59:34 and 12:59:34.

**Comment by ScrapMetalGolem** [comment]
The error message I posted earlier didn't format correctly and is missing the first line:

```12:59:34 [WARN] [BanFromClaim] Task #124 for BanFromClaim v1.7.2 generated an exception
java.lang.NullPointerException: Cannot invoke "com.griefdefender.api.claim.Claim.isWilderness()" because "claim" is null
        at Banfromclaim.jar/no.vestlandetmc.BanFromClaim.hooks.GriefDefenderHook.regionExist(GriefDefenderHook.java:104) ~[Banfromclaim.jar:?]
        at Banfromclaim.jar/no.vestlandetmc.BanFromClaim.config.ClaimData.cleanDatafile(ClaimData.java:173) ~[Banfromclaim.jar:?]
        at Banfromclaim.jar/no.vestlandetmc.BanFromClaim.BfcPlugin$1.run(BfcPlugin.java:76) ~[Banfromclaim.jar:?]
        at org.bukkit.craftbukkit.scheduler.CraftTask.run(CraftTask.java:86) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at org.bukkit.craftbukkit.scheduler.CraftScheduler.mainThreadHeartbeat(CraftScheduler.java:475) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1762) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:509) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1634) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1334) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:333) ~[purpur-1.21.jar:1.21-2283-75e81d8]
        at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]```

**Comment by Baktus79** [comment]
Could you try to delete the banfromclaim folder and turn on the server again?

**Comment by Baktus79** [comment]
Forget what I said, I need to investigate. GriefDefender might have changed their API.

**Comment by Baktus79** [comment]
Fixed in the latest release.

**Comment by ScrapMetalGolem** [comment]
Fixed in 1.7.3, thanks!
